CodeGeneration/Sources/SyntaxSupport/TypeNodes.swift

```diff
@@ -604,4 +604,25 @@ public let TYPE_NODES: [Node] = [
     nameForDiagnostics: nil,
     elementChoices: [.simpleTypeSpecifier, .lifetimeTypeSpecifier]
   ),
+
+  Node(
+    kind: .genericArgumentType,
+    base: .type,
+    nameForDiagnostics: "generic argument type",
+    children: [
+      Child(
+        name: "value",
+        kind: .nodeChoices(choices: [
+          Child(
+            name: "type",
+            kind: .node(kind: .type)
+          ),
+          Child(
```

Comment:
Could you create an experimental feature for value generics and mark the `expr` case as `experimentalFeatures: [.yourFeature]`?